sequencefile/header.go

```diff
@@ -0,0 +1,145 @@
+package sequencefile
+
+import (
+	"encoding/binary"
+	"fmt"
+)
+
+// A Header represents the information contained in the header of the
+// sequencefile.
+type Header struct {
+	Version                   int
+	Compression               compression
+	CompressionCodecClassName string
+	KeyClassName              string
+	ValueClassName            string
+	Metadata                  map[string]string
+	SyncMarker                string
+}
+
+// ReadHeader reads the Header from the input stream, and fill in the Header
+// struct with the values. This should be called when the reader is positioned
+// at the start of the file or input stream, before any records are read.
+//
+// ReadHeader will also validate that the settings of the sequencefile
+// (version, compression, key/value serialization, etc) are compatible.
+func (r *Reader) ReadHeader() error {
+	magic, err := r.consume(4)
+	if err != nil || string(magic[:3]) != "SEQ" {
+		return fmt.Errorf("Error reading magic number: %s", magic)
```

Comment:
This is slightly misleading if `err != nil` since the message talks about the magic.